spacy/pipeline/entity_linker.py

```diff
@@ -90,6 +93,22 @@ def make_entity_linker(
         produces a list of candidates, given a certain knowledge base and a textual mention.
     scorer (Optional[Callable]): The scoring method.
     """
+    # Handle legacy model
+    if model.name == "spacy.EntityLinker.v1":
```

Comment:
How did you verify that this works? I don't really see how it could, the `model.name` is never set to that specific string as far as I can see.

Reply:
I think I completely overlooked this - it's clear looking at it now it won't work at all, since `model.name` is the automatically constructed Thinc name. 

However looking over this I'm not sure the architecture name is available at all at this point. It's not in the model, it's not in the arguments here, and `nlp.config` doesn't seem to contain the config for the current component (and peeking at that would be nasty anyway). 

I need to think about this more, but can we even check the architecture here?

Reply:
My solution for the issue here is to look for the name of the new custom component that pulls out the spans in the thinc model name. If it's not there, I assume it's the old model. 

That feels awfully messy, but I think it's the cleanest way to check things here, since the model name isn't actually available.

Reply:
It would work for now, but the solution does feel a bit brittle & implicit, making it prone for bugs in the future.

I'm almost tempted to define a new `attr` that would record the factory, maybe something like
```
model.attrs["factory"] = {"EntityLinker": 2}
```
And then any internal check could verify the version, and if the right attribute is not available, assume it's below whatever you're checking, because it's a legacy one. In time I think this might be the most robust/clear solution, but it would require maintenance and making sure things stay consistent whenever we introduce a new model version...